### ISSUE TITLE
Poll PIDs of child processes to detect OOM killing

### ISSUE THREAD

**Issue by philippotto** [scalableminds/webknossos-libs#552]
When using multiprocessing (via clustertools), it can happen that the child processes are killed due to OOM. If the OOM killer decides to kill the children, the parent potentially waits forever (sometimes a broken process pool exception will be raised, but we've seen cases where this does not happen). A workaround could be to regularly poll for the child PIDs. If they are not running, a warning could be emitted. Maybe a hard timeout would also make sense (where the main process exists if the timeout is exceeded).

**Comment by fm3** [comment]
I agree that a hard timeout makes sense, this way we can get a job notification in slack (not sure how easy or reasonable it would be to implement the warning also via slack, but not hard kill)

**Comment by normanrz** [comment]
I think a hard timeout would be hard to define. There can be valid cases for jobs that go on for days.

**Comment by philippotto** [comment]
Update: When creating https://github.com/scalableminds/webknossos-libs/pull/739 I noticed that OOMs typically end in a `BrokenProcessPool` exception. That exception can (and will be) handled by the resumable executor in the upcoming version of vx. I think, older python versions sometimes were not able to catch the broken process pool (ending up in a hanging state), but I think I haven't seen this for quite a while (last time was https://github.com/scalableminds/webknossos-libs/issues/539, but I'm not sure which python version was used there). So, I'd defer this issue for now until we see cases where the `BrokenProcessPool` exception is not triggered.
